be/src/util/mem_info.cpp

```diff
@@ -115,43 +115,47 @@ std::string MemInfo::debug_string() {
 void MemInfo::set_memlimit_if_container() {
     // check if application is in docker container or not via /.dockerenv file
     bool running_in_docker = fs::path_exist("/.dockerenv");
-    if (running_in_docker) {
-        struct statfs fs;
-        int err = statfs("/sys/fs/cgroup", &fs);
-        if (err < 0) {
-            LOG(WARNING) << "Fail to get file system statistics. err: " << errno_to_string(err);
-            return;
-        }
+    if (!running_in_docker) {
+        return;
+    }
+    struct statfs fs;
+    int err = statfs("/sys/fs/cgroup", &fs);
+    if (err < 0) {
+        LOG(WARNING) << "Fail to get file system statistics. err: " << errno_to_string(err);
+        return;
+    }
 
-        std::ifstream memoryLimit;
+    StringParser::ParseResult result = StringParser::PARSE_FAILURE;
+    if (fs.f_type == TMPFS_MAGIC) {
+        // cgroup v1
+        // Read from /sys/fs/cgroup/memory/memory.limit_in_bytes
+        std::ifstream ifs;
         std::string line;
-        StringParser::ParseResult result = StringParser::PARSE_SUCCESS;
-        if (fs.f_type == TMPFS_MAGIC) {
-            // cgroup v1
-            // Read from /sys/fs/cgroup/memory/memory.limit_in_bytes
-            memoryLimit.open("/sys/fs/cgroup/memory/memory.limit_in_bytes");
-            getline(memoryLimit, line);
-            _s_physical_mem = StringParser::string_to_int<int64_t>(line.data(), line.size(), &result);
-        } else if (fs.f_type == CGROUP2_SUPER_MAGIC) {
-            // cgroup v2
-            // Read from /sys/fs/cgroup/memory/memory.max
-            memoryLimit.open("/sys/fs/cgroup/memory.max");
-            getline(memoryLimit, line);
-
-            if (line == "max") {
-                _s_physical_mem = std::numeric_limits<int64_t>::max();
-            } else {
-                _s_physical_mem = StringParser::string_to_int<int64_t>(line.data(), line.size(), &result);
-            }
+        ifs.open("/sys/fs/cgroup/memory/memory.limit_in_bytes");
+        if (!ifs.good() || !ifs.is_open()) {
+            return;
         }
-
-        if (result != StringParser::PARSE_SUCCESS) {
-            _s_physical_mem = std::numeric_limits<int64_t>::max();
+        getline(ifs, line);
```

Comment:
suggest to implement a utility function (readBytes/readFully) to read the entire content from a file. referencing to java IO guava, io-common.

Reply:
updated